components/places/src/db/db.rs

```diff
@@ -232,16 +232,25 @@ mod sql_fns {
     // #[inline(never)] ensures they show up in profiles.
 
     #[inline(never)]
-    pub fn hash(ctx: &Context) -> rusqlite::Result<i64> {
+    pub fn hash(ctx: &Context) -> rusqlite::Result<Option<i64>> {
         Ok(match ctx.len() {
             1 => {
-                let value = get_raw_str(ctx, "hash", 0)?;
-                hash::hash_url(value)
+                // This is a deterministic function, which means sqlite
+                // does certain optimizations which means hash() may be called
+                // with a null value even though the query prevents the null
+                // value from actually being used. As a special case, we return
+                // null when the input is NULL. We return NULL instead of zero
+                // because the hash columns are NOT NULL, so attempting to
+                // actually use the null should fail.
+                match get_raw_opt_str(ctx, "hash", 0)? {
```

Comment:
I _think_ you can tighten this up a bit, if you want:

```suggestion
                get_raw_opt_str(ctx, "hash", 0)?.map(|value| hash::hash_url(value) as i64)
```

...But it's your call.

Reply:
nice, thanks!